### PR TITLE
docs: Search Metadata Batch 03

### PR DESCRIPTION
**Pull request by boruszak**
*This PR was created as part of the Consul Docs DevDot migration.*

## Consul Docs Search Refresh | Batch 03

Page metadata headers in the Consul documentation currently include “page title” and “page description” fields. Because our search functions do not match full article text, these fields are used to: 
1. Provide the terms that the internal search functions match with to produce results, meaning that these descriptions index the site’s content for search
1. Populate the internal search results, meaning that these descriptions will be displayed to users as a way to help them select the most relevant page
1. Order internal search results based on Title match and then Description match, alphabetizing within each, meaning that the words used in the page titles have a higher impact on the search results that appear first

Additionally, these fields describe pages for their external Google Search results, meaning that the first 160 characters of each description greatly impact search rankings and user decisions when selecting pages.

## Notes for Reviewers
- **Page titles** currently apply ad hoc punctuation schema in an attempt to include the most important terms and produce more meaningful top results when assuming results are returned alphabetically. Please remember that titles _will continue to change_ as we make organizational and display adjustments in DevDot. When making suggestions, I encourage you to propose different combinations of terms and punctuation as you see fit - but please be aware that making these suggestions on every single title will not be an efficient use of time. I will do my best to standardize title formatting based on any additional input you provide.
  - **Primary goals in title reviews**: 
    - Accurate categorization. For example, the title has “observability” when it should be “traffic management”
    - Highest priority terms and acronyms used
- **Page descriptions**: Descriptions were written to summarize a page’s content with an eye towards how components work in Consul, user end-goals, and what content is actually contained in the page. Additionally, every description’s first 160 characters include a complete sentence, and the character cut-off was considered semantically significant for each description (for example, if a word is cut off, it should still be comprehensible, so there aren’t lingering individual characters). _You do not have to consider character counts in your review_. Make suggestions, and I will review them in light of these constraints to fine-tune wording.
  - **Primary goals in description reviews**: 
    - Technical accuracy in descriptions
    - Includes terms that could reasonably be part of to find this page’s content
    - Does not use words that are semantically insignificant or misleading. For example, “get” is not a descriptive verb *and* it has a specific use in API requests, so it has increased potential to skew search results.

### Affected Directories
This batch contains refreshed headers for all pages in the following directories:
- /docs/internals
- /docs/architecture
- /docs/concepts
- /docs/connect/registration
- /docs/connect/native
- /docs/connect/l7-traffic
- /docs/connect/ca
- /docs/connect/observability

### Branch Organization
To make review more manageable, these PRs are being created and submitted in batches. Batches use branch names formatted as: `docs/search-desc-batchXX`. These branches are set to merge into a single branch named `docs/search-metadata-headers` so that all updates can be merged into the `dev-portal` branch simultaneously if desired.

Please direct any questions to @boruszak.

### CODE REVIEW

File: website/content/docs/connect/registration/service-registration.mdx [hunk 1]
```diff
@@ -1,13 +1,11 @@
 ---
 layout: docs
-page_title: Connect - Service Registration
+page_title: Service Mesh Proxy: Configuration
```
Comment (jkirschner-hashicorp) [inline website/content/docs/connect/registration/service-registration.mdx:3]: I don't like how clunky the title I suggested is, but it's a more accurate reflection of what is going on (and how it's different from the alternative).
```suggestion
page_title: Service Mesh Proxy Registration: Separate from Service Registration
```

File: website/content/docs/connect/registration/sidecar-service.mdx [hunk 2]
```diff
@@ -1,12 +1,11 @@
 ---
 layout: docs
-page_title: Connect - Sidecar Service Registration
-description: |-
-  Sidecar service registrations provide a convenient shorthand for registering a
-  sidecar proxy inline with a regular service definition.
+page_title: Service Mesh Proxy: Sidecar Configuration
```
Comment (jkirschner-hashicorp) [inline website/content/docs/connect/registration/sidecar-service.mdx:3]: I don't like how clunky the title I suggested is, but it's a more accurate reflection of what is going on (and how it's different from the alternative).
```suggestion
page_title: Service Mesh Proxy Registration: Included in Service Registration
```

File: website/content/docs/connect/registration/service-registration.mdx [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 description: >-
-  A per-service proxy sidecar transparently handles inbound and outbound service
-  connections. You can register these sidecars with reasonable defaults by nesting
-  their definitions in the service definition.
+Use service mesh proxy configuration files to define the parameters and behaviors of custom proxies deployed with services in your mesh. Learn about configuration options and how to format them with examples.
```
Comment (jkirschner-hashicorp) [inline website/content/docs/connect/registration/service-registration.mdx:5]: This page is confusing (in my opinion), but isn't about custom proxies specifically. You can use this with Envoy proxies.
```suggestion
You can register a service mesh sidecar proxy separate from the registration of the service instance it fronts. Learn about proxy configuration options and how to format them with examples.
```

File: website/content/docs/connect/registration/sidecar-service.mdx [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
+description: >-
+In a service mesh, sidecar proxies run in the same node as service instances and are configured in service definition files. Learn about default settings, customizable parameters, limitations, and lifecycle behaviors.
```
Comment (jkirschner-hashicorp) [inline website/content/docs/connect/registration/sidecar-service.mdx:5]:
```suggestion
You can conveniently register a service instance and its sidecar proxy together. Learn about default settings, customizable parameters, limitations, and lifecycle behaviors of the sidecar proxy.
```

File: website/content/docs/connect/registration/sidecar-service.mdx [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
 ---
 
-# Sidecar Service Registration
+# Service Mesh Proxy Sidecar Configuration
```
Comment (jkirschner-hashicorp) [inline website/content/docs/connect/registration/sidecar-service.mdx:8]: Same thoughts as the `page_title` comment above

File: website/content/docs/internals/index.mdx [hunk 6]
```diff
@@ -1,12 +1,11 @@
 ---
 layout: docs
-page_title: Internals
+page_title: Internals Overview
 description: >-
-  This section covers some of the internals of Consul, such as the architecture,
-  consensus and gossip protocols, and security model.
+Consul is a complex system that relies on a number of protocols and concepts to work. Learn about Consul's internal properties and how it works under the hood.
```
Comment (jkirschner-hashicorp) [inline website/content/docs/internals/index.mdx:5]: I don't think we want to focus on complexity. Ideally, that's abstracted most of the time. The goal is to give more advanced knowledge where it's needed to promote better/more advanced usage.
```suggestion
To enhance your mental model of how to use, troubleshoot, and secure Consul, learn more about Consul's internal properties and how it works under the hood.
```

**Review by jkirschner-hashicorp — COMMENTED** [review]
@boruszak : review 3 of 4 completed! Same types of feedback as on reviews 1-2.

**Comment by lkysow** [comment]
Again I won't take a look unless you need me to as well as Jared.

**Pull request merged**